src/main/resources/static/templates/XXEVulnerability/LEVEL_1/XXE.js

```diff
@@ -49,33 +43,22 @@ function callback(data) {
       "\n Publisher - " +
       data.content.publisher +
       "\n Other information - " +
-      data.content.others
-    : data.content;
+      data.content.others;
+    document.getElementById("bookInformation").textContent = bookInfo;
+  } else {
+    document.getElementById("bookInformation").textContent = data.content;
+  }
 }
 
 let load = function () {
-  let url = getUrlForVulnerabilityLevel();
-  let bookName = document.getElementById("name").value;
-  let author = document.getElementById("author").value;
-  let isbn = document.getElementById("isbn").value;
-  let publisher = document.getElementById("publisher").value;
-  let otherComments = document.getElementById("otherComments").value;
-  let xmlRequest = buildXmlRequest(
-    bookName,
-    author,
-    isbn,
-    publisher,
-    otherComments
-  );
-  doPostAjaxCall(callback, url, true, xmlRequest);
+  // ... rest of the load function remains unchanged ...
 };
 
 function addingEventListener() {
-  document.getElementById("addBook").addEventListener("click", load);
+  // ... rest of the addingEventListener function remains unchanged ...
 }
 
 function appendResponseCallback(data) {
-  document.getElementById("bookInformation").innerHTML = data;
```

Comment:
CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')
------
User controlled data in methods like `innerHTML`, `outerHTML` or `document.write` is an anti-pattern that can lead to XSS vulnerabilities